actors/runtime/src/runtime/fvm.rs

```diff
@@ -263,28 +261,45 @@ where
         value: TokenAmount,
     ) -> Result<RawBytes, ActorError> {
         if self.in_transaction {
-            return Err(actor_error!(SysErrIllegalActor; "runtime.send() is not allowed"));
+            return Err(actor_error!(user_assertion_failed; "runtime.send() is not allowed"));
         }
-        // TODO: distinguish between "syscall" errors and actor exit codes.
         match fvm::send::send(&to, method, params, value) {
             Ok(ret) => {
                 if ret.exit_code.is_success() {
                     Ok(ret.return_data)
                 } else {
-                    Err(ActorError::from(ret.exit_code))
+                    // The returned code can't be simply propagated as it may be a system exit code.
+                    // TODO: improve propagation once we return a RuntimeError.
+                    // Ref https://github.com/filecoin-project/builtin-actors/issues/144
+                    Err(actor_error!(
+                        user_assertion_failed,
+                        format!(
+                            "send to {} method {} aborted with code {}",
+                            to, method, ret.exit_code
+                        )
+                    ))
                 }
             }
             Err(err) => Err(match err {
+                // Some of these errors are from operations in the Runtime or SDK layer
+                // before or after the underlying VM send syscall.
                 ErrorNumber::NotFound => {
-                    actor_error!(SysErrInvalidReceiver; "receiver not found")
+                    // How do we know the receiver is what triggered this?
+                    // An error number doesn't carry enough information at this level
+                    // above the raw syscall (and even then).
```

Comment:
As noted on the other PR, the syscall is defined to return "not found" when the receiver isn't found. Maybe the system has a bug and will end up returning not found in other cases, but then we have a bug?